tarantool_test.go

```diff
@@ -1216,60 +1144,38 @@ func TestClientSessionPush(t *testing.T) {
 	}
 
 	for i := 0; i < len(its); i++ {
-		pushCnt := uint64(0)
-		respCnt := uint64(0)
-
 		it = its[i]
 		for it.Next() {
-			resp = it.Value()
+			resp := it.Value()
 			if resp == nil {
 				t.Errorf("Response is empty after it.Next() == true")
 				break
 			}
-			if resp.Pos != nil {
-				t.Errorf("Response should not have a position")
+			data, err := resp.Decode()
+			if err != nil {
+				t.Errorf("Failed to Decode: %s", err.Error())
+				break
 			}
-			if len(resp.Data) < 1 {
+			if len(data) < 1 {
 				t.Errorf("Response.Data is empty after CallAsync")
 				break
 			}
-			if resp.Code == PushCode {
-				pushCnt += 1
-				val, err := test_helpers.ConvertUint64(resp.Data[0])
-				if err != nil || val != pushCnt {
-					t.Errorf("Unexpected push data = %v", resp.Data)
-				}
-			} else {
-				respCnt += 1
-				val, err := test_helpers.ConvertUint64(resp.Data[0])
-				if err != nil || val != pushMax {
-					t.Errorf("Result is not %d: %v", pushMax, resp.Data)
-				}
-			}
```

Comment:
Please, return the deleted code and use `it.IsPush()` instead of `resp.Code == PushCode`